cropro.py

```diff
@@ -122,14 +138,20 @@ def initUI(self):
         self.currentProfileDeckCombo = QComboBox()
         currentProfileDecks = mw.col.decks.all()
         currentProfileDecks.sort(key=lambda d: d['name'])
-        selectedDeckId = mw.col.decks.selected()
         selectedIndex = None
-        for idx, deck in enumerate(currentProfileDecks):
-            self.currentProfileDeckCombo.addItem(deck['name'], deck['id'])
-            if deck['id'] == selectedDeckId:
-                selectedIndex = idx
-        if selectedIndex is not None:
-            self.currentProfileDeckCombo.setCurrentIndex(selectedIndex)
+        if DEFAULT_DECK_DESTINATION != '':
+            for idx, deck in enumerate(currentProfileDecks):
+                self.currentProfileDeckCombo.addItem(deck['name'], deck['id'])
+                if deck['name'] == DEFAULT_DECK_DESTINATION:
+                    selectedIndex = idx
+        else:
+            selectedDeckId = mw.col.decks.selected()
+            for idx, deck in enumerate(currentProfileDecks):
+                self.currentProfileDeckCombo.addItem(deck['name'], deck['id'])
+                if deck['id'] == selectedDeckId:
+                    selectedIndex = idx
```

Comment:
Consider using something along the lines of this to simplify implementation:
```
        field = 'name' if DEFAULT_DECK_DESTINATION != '' else 'id'
        comparison_value = DEFAULT_DECK_DESTINATION if DEFAULT_DECK_DESTINATION != '' else mw.col.decks.selected()
        for idx, deck in enumerate(currentProfileDecks):
            self.currentProfileDeckCombo.addItem(deck['name'], deck['id'])
            if deck[field] == comparison_value:
                selectedIndex = idx
```
Also, I am not a fan of that `addItem` part, does not seem efficient. Python has some kick-ass features, consider cleaning it up

Reply:
Hold on, I am not even sure if this works correctly. It seems what you are trying to do is some fancy "find" operation but are not terminating the loop when the desired `selectedIndex` is found